node/node.go

```diff
@@ -81,13 +82,24 @@ var (
 
 type pubSubProcessorFunc func(ctx context.Context, msg pubsub.Message) error
 
+type nodeChainReader interface {
```

Comment:
Non-Blocking: It's great that we're moving away from the `ChainStore` java style interface. It would be nice if this PR actually remove it, rather than leaving  both styles in the codebase, but it appears it's been replaced enough times to set the pattern. It would also be better to use the new consumer defined interfaces to stub or fake testing, but this is a step in the right direction.

Reply:
That was sort of a side-goal of this PR, but I chose not to do it yet to keep the diff size down. I think it would be a great follow up task.